src/foam/nanos/notification/email/SMTPEmailService.js

```diff
@@ -191,25 +213,42 @@ foam.CLASS({
     },
     {
       name: 'sendEmail',
-      javaCode:
-      `
-      OMLogger omLogger = (OMLogger) x.get("OMLogger");
-      Logger logger = (Logger) getX().get("logger");
+      javaCode: `
+        emailMessage = (EmailMessage) emailMessage.fclone();
+        MimeMessage message = createMimeMessage(emailMessage);
+        Logger logger = (Logger) getX().get("logger");
         try {
-          omLogger.log("Pre send email request");
-          MimeMessage message = createMimeMessage(emailMessage);
-          Transport transport = getSession_().getTransport("smtp");
-          transport.connect();
-          logger.info("SMTPEmailService connected.");
-          transport.send(message, getUsername(), getPassword());
-          logger.info("SMTPEmailService sent MimeMessage.");
-          transport.close();
-          logger.info("SMTPEmailService finish.");
-          omLogger.log("Post send email request");
-        } catch (Exception e) {
-          logger.error("SMTPEmailService failed to finish. " + e);
+          getTransport_().send(message);
+          emailMessage.setStatus(Status.SENT);
+          logger.debug("SMTPEmailService sent MimeMessage.");
+        } catch ( SendFailedException e ) {
+          emailMessage.setStatus(Status.FAILED);
+          logger.error("SMTPEmailService sending MimeMessage failed. " + e);
+        } catch ( MessagingException e ) {
+          try {
+            getTransport_().close();
+          } catch ( Exception e2 ) {
+            logger.error("Failed to close transport. " + e2);
+          }
+          clearTransport_();
+          logger.error("SMTPEmailService sending MimeMessage failed. " + e);
+        }
+        return emailMessage;
+      `
+    },
+    {
+      name: 'execute',
+      javaCode: `
+        // TODO: This should be a cron job.
```

Comment:
We need a cron job to either call this service.execute, or move this to the cron job itself.  Moving this makes most sense.